### ISSUE TITLE
Version 1 header is wrong.

### ISSUE THREAD

**Issue by tatarize** [JoshVarga/libpes#2]
```
header->hoopSize = binaryReadInt16(file); // 0 = 100x100 else 130x180 or above
header->otherHoopData = binaryReadByte(file);
header->otherHoopData2 = binaryReadByte(file);
```

These are properly hoopSize = 100x100
Use Existing Design Area.
[And number of CSewSeg objects.](https://github.com/frno7/libpes/wiki/PES-CSewSeg-section)

One could properly call that second item "otherHoopData" but the last element there is quite critical. It's the count of CSewSeg segment blocks.

**Comment by tatarize** [comment]
Actually it turns out the value is after that otherhoopdata value and still critical and missing. Though not that important it's the number of objects in the pes document. Whether sewsegs or rectangles or circles etc. It's actually the last value in all the headers.

**Comment by JoshVarga** [comment]
ok, great, I will check it out and update the code. Has this code been beneficial so far? It took a bunch of trial and error with many different files to figure it out this far.

**Comment by tatarize** [comment]
Yeah, quite helpful. I've been working on getting my stuff to load and save PES v.6. My photoembroidery stuff is highly color dependent and it would be nice to get more exacting threads colors as part of the output. Tweaked the graphics too, so I'm getting some more use of your pesThumb utility. 

I did hit a quirk with regard to Wilcom's software. Turns out the PES thread colors are supposed to be unique and map to the specific number reference in the PEC section. So 10, 5, 10, 29, 5 would be assigned thread 1, thread 2, (reuse thread 1), thread 3, (reuse thread 2). This is the way Brother expects them, and writes them. But, Wilcom's code base (so all Wilcom products, a lot of conversion tools) all prefer a 1 to 1 thread listing to colors in PEC. This weirdly results in weird incompatibilities. Wilcom loads some PES files wrong. And produces some invalid PES files (2 different colors from different threads get pigeon-holed into the same index) that don't display correctly in PE-Design's  Open dialog Preview and some other conversions.

So in order to maximize compatibility of the files I produce, I need to never reuse the index color values and provide a 1 to 1 thread listing. Because only that is fully compatible with both Wilcom and Brother. And I have a weird hybrid load routine, if the colors in the header is equal to the colors in the PEC it treats them as 1:1 threads to colors (regardless whether any index is reused or not) and if they are different it assigns them in a unique fashion as Brother apparently intended, and if there were no colors in the header it uses the index colors.

It's weirdly complex to deal with a bug in Wilcom, and involves pushing colors to progressively less similar indexes.

---

The biggest time suck was writing the output of V6 as it has another block of order/group data that occurs right before the PEC block. It apparently is the number of objects and how they are arranged and grouped in Sewing Order / Color. Though the default values are pretty easy to create, once I figured out what missing data looked like (though I have little clue what they mean, other than being the sew-order and grouping).

Also, the reader you gave for programmable pattern fill doesn't work, so I just decided to abort if I ever see a value other than 0 for those elements, which would be extremely rare. But, I can totally read and write version 6 with max compatibility, and thus with full gamut thread.

[tree.zip](https://github.com/JoshVarga/libpes/files/1466355/tree.zip)

**Comment by tatarize** [comment]
Though actually, I am pretty sure bytes 0E-0F are actually "01" for PE-Design and "02" for Art, Wilcom etc.  I have a file in 5.5 (same header as 5), which says "10" there. They are always actual text numbers. I'm thinking it might be some kind of manufacturer code and be actual text for the user looking at any file after V.3. There's no function left that it could be referring to and setting them to any value makes no difference for any software.  But, notably all the Wilcom/Art stuff says 02 there. But, if I can detect Wilcom created files I could work around that bug rather effectively. 

And the third 100 in v. 6 0x64 does nothing except it is preserved and throws an error if you set it too low when you're messing with the DesignWidth and DesignHeight it will object if you hexedited the value to something like 0x17 even though there's literally no value outside the given range for that value.

**Comment by tatarize** [comment]
*sigh* no such luck, both programs make 01 for version 4. Might just be an extra bit of version code or something. Pretty sure it's just text tho.
